File: android/lib/services/subscription_url_policy.dart
/// Policy for remote subscription fetch URLs (HTTPS by default).
class SubscriptionUrlPolicy {
  static const int maxRedirects = 5;

  /// Accepts https anywhere, or http only to loopback development hosts.
  static bool isSecure(Uri? uri) {
    if (uri == null || !uri.hasAuthority) return false;
    final scheme = uri.scheme.toLowerCase();
    if (scheme == 'https') return true;
    if (scheme != 'http') return false;
    final host = uri.host.toLowerCase();
    return host == 'localhost' ||
        host == '127.0.0.1' ||
        host == '::1' ||
        host == '[::1]';
  }

  static bool isRedirectStatus(int statusCode) =>
      statusCode == 301 ||
      statusCode == 302 ||
      statusCode == 303 ||
      statusCode == 307 ||
      statusCode == 308;
}

File: ios/App/OpenPPP2/SubscriptionUrlPolicy.swift
import Foundation

/// Policy for remote subscription fetch URLs (HTTPS by default).
public enum SubscriptionUrlPolicy {
    public static let maxRedirects = 5

    /// Accepts https anywhere, or http only to loopback development hosts.
    public static func isSecure(_ url: URL) -> Bool {
        guard let scheme = url.scheme?.lowercased() else { return false }
        if scheme == "https" { return true }
        if scheme != "http" { return false }
        guard let host = url.host?.lowercased() else { return false }
        return host == "localhost" ||
            host == "127.0.0.1" ||
            host == "::1" ||
            host == "[::1]"
    }
}
